rust/rope/src/diff.rs

```diff
@@ -245,21 +271,82 @@ impl DiffBuilder {
     }
 }
 
+/// An interval for a specific rope.
+#[derive(Debug, Clone)]
+pub struct RopeSlice<'a> {
+    rope: &'a Rope,
+    range: Range<usize>,
+}
+
+impl PartialEq for RopeSlice<'_> {
+    fn eq(&self, other: &Self) -> bool {
+        let mut it1 =
+            self.rope.iter_chunks(Interval::from(self.range.clone())).flat_map(|x| x.chars());
```

Comment:
ditto here, and `chars`; we want to use `bytes`.

I would just add a method to `RopeSlice`, like

```rust
impl<'a> RopeSlice<'a> {
    fn iter_bytes(&'a self) -> impl Iterator<Item=u8> + 'a {
            self.rope.iter_chunks(self.range.clone()).flat_map(|chunk| chunk.bytes())
        }
}
```